### ISSUE TITLE
Improve interface->component links

### ISSUE THREAD

**Issue by IoannisP-ITENG** [faebryk/faebryk#50]
### Feature Request

Currently we need to explicitly add this relationship which is quite error prone.
We need a way to add this relationship more elegantly while keeping the possibility of interfaces to be free from components.

Example of how it looks currently: samples/resistors_and_nand.py#22:
```
  # power
  battery = Component()
  battery.power = Power()
  battery.add_trait(has_defined_interfaces([battery.power]))
  battery.power.set_component(battery)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by IoannisP-ITENG** [comment]
Maybe with the has_interfaces trait something?

**Comment by IoannisP-ITENG** [comment]
Fixed in #121
